### PR TITLE
[Merged by Bors] - chore(Data/ENat): unbundle `ENat.toNat`

### PR DESCRIPTION
**Pull request by Rida-Hamadani**
This will allow us to write `n.toNat` instead of `ENat.toNat n`

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/ENat/Basic.lean [hunk 1]
```diff
@@ -96,13 +96,20 @@ instance : WellFoundedRelation ℕ∞ where
...
+def toNatHom : MonoidWithZeroHom ℕ∞ ℕ where
   toFun := WithTop.untop' 0
   map_one' := rfl
   map_zero' := rfl
   map_mul' := WithTop.untop'_zero_mul
-#align enat.to_nat ENat.toNat
+#align enat.to_nat ENat.toNatHom
+
+/-- Conversion of `ℕ∞` to `ℕ` sending `∞` to `0`. -/
+def toNat : ℕ∞ → ℕ := toNatHom
```
Comment (YaelDillies) [inline Mathlib/Data/ENat/Basic.lean:108]: `toNat` should be defined before `toNatHom`

File: Mathlib/Data/ENat/Basic.lean [hunk 2]
```diff
@@ -96,13 +96,20 @@ instance : WellFoundedRelation ℕ∞ where
   rel := (· < ·)
   wf := IsWellFounded.wf
 
-/-- Conversion of `ℕ∞` to `ℕ` sending `∞` to `0`. -/
-def toNat : MonoidWithZeroHom ℕ∞ ℕ where
+/-- Homomorphism from `ℕ∞` to `ℕ` sending `∞` to `0`. -/
+def toNatHom : MonoidWithZeroHom ℕ∞ ℕ where
   toFun := WithTop.untop' 0
```
Comment (YaelDillies) [inline Mathlib/Data/ENat/Basic.lean:101]:
```suggestion
  toFun := toNat
```

File: Mathlib/Data/ENat/Basic.lean [hunk 3]
```diff
@@ -96,13 +96,20 @@ instance : WellFoundedRelation ℕ∞ where
...
   map_mul' := WithTop.untop'_zero_mul
-#align enat.to_nat ENat.toNat
+#align enat.to_nat ENat.toNatHom
+
+/-- Conversion of `ℕ∞` to `ℕ` sending `∞` to `0`. -/
+def toNat : ℕ∞ → ℕ := toNatHom
+
+@[simp]
+theorem toNatHom_eq_toNat : toNatHom n = toNat n :=
+  rfl
```
Comment (YaelDillies) [inline Mathlib/Data/ENat/Basic.lean:112]:
```suggestion
@[simp, norm_cast] lemma coe_toNatHom : toNatHom = toNat := rfl
lemma toNatHom_apply (n : ℕ) : toNatHom n = toNat n := rfl
```

Comment (Rida-Hamadani) [reply Mathlib/Data/ENat/Basic.lean]: Btw, `by norm_cast` can prove `coe_toNatHom`, is it okay to give it `norm_cast` attribute if it is solved by it?

**Comment by b-mehta** [comment]
bors merge

**Pull request closed without merging**